### PR TITLE
Add support for NETStandard and baseline references

### PR DESCRIPTION
**Pull request by ericstj**
This change adds a couple opt-in features to packaging:
1. Using NETStandard instead of DotNet as the portable generation moniker.
2. Raising dependencies to a minimal baseline version.

We need both of these to switch our packages to NETStandard.  
#2 is required because the old packages do not have NETStandard
assets for compile time and NETStandard is not compatible with 
dotnet assets.

This also updates the default minClientVersion to 3.4.0 since we
depend on changes in 3.4.0: the latest generation mapping of
dotnet5.6 > dnxcore50 & UAP and NETStandard support.

/cc @weshaggard

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/Generations.cs [hunk 1]
```diff
@@ -185,29 +195,29 @@ public Version DetermineGenerationFromSeeds(string assemblyName, Version assembl
 
             if (lowerGeneration == null && upperGeneration == null)
             {
-                log.LogMessage(LogImportance.Low, "Assembly {0} is not tracked by generations.", assemblyName);
+                log.LogMessage(LogImportance.Low, $"Assembly {assemblyName} is not tracked by generations.");
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/Generations.cs]: C# 6.0 features :) Hopefully that doesn't break anything, I don't think it will.

**Comment by weshaggard** [comment]
LGTM - Please squash as the commits seem to be going back and forth :)

When do you plan to enable this?

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/Generations.cs (hunk 1 above)
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/Generations.cs]: Buildtools supports c#6.  I'm already using it in the same file.

**Comment by ericstj** [comment]
Sure I will squash.  I am hoping to enable it this week, but I need to get some other things in place first.

**Comment by ericstj** [comment]
Had to update the baseline versions to account for https://github.com/dotnet/corefx/pull/5845

**Pull request merged**